examples/create-your-own-abstractions/somegroup.pp

```diff
@@ -0,0 +1,29 @@
+define somegroup(
+  $ami,
+  $region,
+  $type = 't1.micro',
+  $ensure = 'present',
+) {
+  # For a more complex example we could validate the
+  # arguments using stdlib or the new future parser
+  ec2_instance { ["${title}-1", "${title}-2"]:
+    ensure          => $ensure,
+    region          => $region,
+    image_id        => $ami,
+    security_groups => "${title}-sg",
+    instance_type   => $type,
+  }
+
+  ec2_securitygroup { "${title}-sg":
+    ensure      => $ensure,
+    description => "group for ${title}",
+    region      => $region,
+  }
+
+  # ec2_instance resources need to be deleted before the
+  # associated security groups
+  if $ensure == 'absent' {
```

Comment:
I assume in the ensure => present case, it is the other way round and the the type does the correct auto require? In example code, this should be explained.